charms/knative-serving/src/charm.py

```diff
@@ -135,13 +140,22 @@ def _template_files(self):
 
     @property
     def _context(self):
+        try:
+            custom_images = parse_image_config(self.model.config[CUSTOM_IMAGE_CONFIG_NAME])
+            custom_images = update_images(
+                default_images=DEFAULT_IMAGES, custom_images=custom_images
+            )
+        except ValueError as err:
+            raise ErrorWithStatus(str(err), BlockedStatus)
+
```

Comment:
Just to keep the context property a bit tidy, I wonder if this could be done outside of the `_context` property and just update the context dict with this information, similar to what is done for `self._otel_collector_relation_data`.